staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -156,6 +156,28 @@ type VirtualMachineInstanceSpec struct {
 	AccessCredentials []AccessCredential `json:"accessCredentials,omitempty"`
 }
 
+func (vmiSpec *VirtualMachineInstanceSpec) UnmarshalJSON(data []byte) error {
+	type VMISpecAlias VirtualMachineInstanceSpec
+	var vmiSpecAlias VMISpecAlias
+
+	err := json.Unmarshal(data, &vmiSpecAlias)
+	if err != nil {
+		return err
+	}
+
+	*vmiSpec = VirtualMachineInstanceSpec(vmiSpecAlias)
```

Comment:
I think you should first mutate the alias and only then assign it at the very end.

Reply:
Done.